scripts/tools/zap_regen_all.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+#
+#    Copyright (c) 2020 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import sys
+import os
+from pathlib import Path
+
+# Check python version
+if sys.version_info[0] < 3:
+    print("Must use Python 3. Current version is " + str(sys.version_info[0]))
+    exit(1)
+
+# Check if we are in top of CHIP folder
+if not str(os.getcwd()).endswith("connectedhomeip"):
```

Comment:
This will fail for some of my checkouts, fwiw; they don't have that naming structure (they all start with "connectedhomeip" but end with some labels indicating the purpose).

Can't we just use our own path, walk up two levels, and work relative to that?  Or at least check whether that matches the cwd instead of hardcoding the name of the entire checkout?

Reply:
Ah right. I did not noticed.

I'm actually in a similar case. *None* of my checkouts have that specific name, I'm always using a custom name depending on the features being worked on for checkouts an even my own fork is named `connectedhomeip-1`

Reply:
@bzbarsky-apple @vivien-apple good to know, I'll push a different approach that would in your cases